### PR TITLE
Issue 1698: Use safe_sleep executable in bash scripts

### PR DESCRIPTION
**Pull request by ruvceskistefan**
Problems were:
- the implementation of `sleep` in the case when both ping and sleep are missing in `safe_sleep` function
- usage of standard `sleep` function instead of `safe_sleep` in bash script

Related issue: #1698

After this change, `safe_sleep` is implemented as separate executable bash script, and scripts `run-helper.sh` and `update.sh` are using that function instead of standard `sleep` function.

### CODE REVIEW

File: src/Misc/layoutroot/run-helper.sh.template [hunk 1]
```diff
@@ -44,17 +28,17 @@ elif [[ $returnCode == 1 ]]; then
     exit 0
 elif [[ $returnCode == 2 ]]; then
     echo "Runner listener exit with retryable error, re-launch runner in 5 seconds."
-    safe_sleep
+    ./bin/safe_sleep.sh 5
```
Comment (fhammerl) [inline src/Misc/layoutroot/run-helper.sh.template:31]: I'm always a bit iffy on relative paths, does is work fine if I start run.sh with an absolute path from a different directory?

Comment (ruvceskistefan) [reply src/Misc/layoutroot/run-helper.sh.template]: i changed to absolute paths and changed the location of the `safe_sleep` script, so it is not in bin folder.

**Comment by Mart-Bogdan** [comment]
How this got merged? This is definitely endless loop.

https://github.com/actions/runner/blob/0ebdf9e83d51b0ab8c168a4869954466f3e2e88f/src/Misc/layoutroot/safe_sleep.sh#L3-L6


**EDIT**.  Oh, SECONDS is autoincremented by bash. didn't know that.

**Comment by elupus** [comment]
Why is this not attempting to yield using a potentially available sleep call on the systems where that is available. Ie close to all systems. Could just do that inside the new loop.

Busy waiting is always really bad.

**Pull request merged**